series: series_test.rb test for strings not arrays

stevensonmt opened the pull request:
Previous test looked for arrays of Ints but exercise prompt in readme called for strings.  This update makes tests look for strings.

stevensonmt commented:
Not sure I understand why it's failing this check.  Looks as though it is checking it against a solution that is based on the previous tests looking for arrays of Ints.  I don't know where it's getting that master solution file from so can't change it.  Obviously that solution was written to pass the previous tests, which are not consistent with the exercise prompt in the readme.

Insti commented:
Thanks @stevensonmt I'm happy to see you worked out how to make a pull request.

The tests are run against an example solution which can be found at: https://github.com/exercism/xruby/blob/master/exercises/series/example.rb

Do you want to try and update that to match your new tests?

You can come and ask questions in the Gitter chatroom: https://gitter.im/exercism/xruby if you want some 'live' help.

stevensonmt commented:
Updated the example.rb so that it now passes the tests when the tests expect substrings instead of arrays of ints.

Insti commented on exercises/series/example.rb:
```diff
@@ -1,27 +1,16 @@
-  def convert_to_digits(s)
-    s.chars.to_a.map(&:to_i)
+    substrings = []
+    until @series.length < n
```
This has the same problem as your original solution, in that it won't work if called twice.

Insti replied:
Maybe it would help if you added an additional test case for this at the end.

Insti commented on exercises/series/series_test.rb:
```diff
@@ -6,87 +6,87 @@
 
   def test_simple_slices_of_one_again
-    skip
+    #skip
```
We need to start with all tests except the first skipped.

Insti commented:
Nice work @stevensonmt this is heading in the right direction.

stevensonmt commented:
@Insti -- Should have caught that about skipping all but the first test.  Added test for sequential calls of slices method and updated the example so that it does not mutate the original string.

Insti commented on exercises/series/series_test.rb:
```diff
@@ -6,92 +6,99 @@
+    #skip
+    series = Series.new('1234')
+    series.slices(2)
+    series.slices(3)
```
Remember to `assert` something. :smile:

kotp commented:
@stevensonmt when you are ready, you can squash the commits for the next review.  There is a way to leave the skips there and still be able to run the tests, which is why I think the tests may have been skipped (it would be a normal reaction while developing).

Here is that command:

`ASSIGNMENT=series make test-assignment`

This will run the tests and automatically skip them, while doing the other needful things.  Not magic, you can read the Makefile to see exactly what is happening.

Insti commented:
I'm still seeing the `skip`s are commented out.

Insti commented:
Great, thanks. Is there anything else you want to do before we merge this @stevensonmt ?

kotp commented:
Looks better.

stevensonmt commented:
I don't think there's anything else to do, but obviously I've a long way to go at recognizing the things that need doing.

Insti commented:
Congratulations @stevensonmt, you're an open source contributor! :medal_sports: 

Thanks a lot for noticing a problem and doing something about fixing it and for your all effort in working out what you needed to do.

There's usually always more to do :smile: , but no one person needs to do it all! :heart:

Insti commented:
PS. It might be a week or two before the change makes it to the live site, live updates are not happening so often over the holiday period. But you will see it there eventually.
